### PR TITLE
Adjust AI spell estimates

### PR DESCRIPTION
**Pull request by idshibanov**
Related to #3239 (partial fix).

Separate tactical and strategic magic estimates to make sure AI's archer strength wouldn't be overvalued. Tune values to account for number of spell casts possible.

### CODE REVIEW

**Comment by ihhub** [comment]
Welcome back, @idshibanov ! 👍

File: src/fheroes2/ai/normal/ai_normal.h [hunk 1]
```diff
@@ -101,6 +101,7 @@ namespace AI
         double getSpellSlowRatio( const Battle::Unit & target ) const;
         double getSpellHasteRatio( const Battle::Unit & target ) const;
         uint32_t spellDurationMultiplier( const Battle::Unit & target ) const;
+        static double CommanderDamageValue( const HeroBase & commander );
```
Comment (ihhub) [inline src/fheroes2/ai/normal/ai_normal.h:104]: Please start method's name from non-capital letter.

Comment (ihhub) [reply src/fheroes2/ai/normal/ai_normal.h:104]: Shall we call it `commanderMaximumSpellDamageValue`? As right now the name is misleading.

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test these changes? The changes are made to adjust estimation by AI towards a castle attacker and overall army strength.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov , I left few styling comments. Could you please check them?

File: src/fheroes2/ai/normal/ai_normal.h (hunk 1 above)
Comment (idshibanov) [reply src/fheroes2/ai/normal/ai_normal.h:104]: > Please start method's name from non-capital letter.

Even for static method?

> Shall we call it commanderMaximumSpellDamageValue? As right now the name is misleading.

A bit too long for my taste, especially `BattlePlanner::commanderMaximumSpellDamageValue()`. Naming things is always hard.

**Comment by Branikolog** [comment]
Hi, @ihhub and @idshibanov !
I'm not sure, what exactly I should test here.
Currently, with this changes in a case we eliminate AI's rangers he's still waiting inside castle, relying on towers, while he can do more efficiently rushing my shooters (since he has considerable army):
image

File: src/fheroes2/ai/normal/ai_normal.h (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/ai/normal/ai_normal.h:104]: It's better to have a long name than a short cryptic one :)

**Comment by idshibanov** [comment]
@Branikolog this case looks like a tower issue rather than spells. Do you still have the save for the siege so I can check it?

**Comment by ihhub** [comment]
Hi @idshibanov and @Branikolog , what we decided to do? Are we good with these changes or do we need to do some modifications?

**Comment by idshibanov** [comment]
> Hi @idshibanov and @Branikolog , what we decided to do? Are we good with these changes or do we need to do some modifications?

I believe that this particular PR is hard to test without siege (and tower strength) changes. I'll probably make the tower change separately then we can come back to this.

**Comment by ihhub** [comment]
@idshibanov , thanks a lot for these changes!

**Pull request merged**